GFDL_tools/fv_ada_nudge.F90

```diff
@@ -1793,26 +1780,29 @@ subroutine fv_ada_nudge_init(time, axes, npz, zvir, ak, bk, ts, phis, gridstruct
                'da ps', 'Pa', missing_value=missing_value )
 
 ! snz add the following lines for recording the return values from the previous assim run
-    call get_mosaic_tile_file(restart_file, restart_file_instance,&
-         & .FALSE., domain)
-
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "u_adj", Atm_var%u_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "v_adj", Atm_var%v_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "t_adj", Atm_var%t_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "q_adj", Atm_var%q_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "ps_adj", Atm_var%ps_adj(:,:), domain=domain)
-
-    if ( file_exist('INPUT/'//trim(restart_file_instance), domain=domain) ) then
-       if ( mpp_pe() .eq. mpp_root_pe() ) then
-          write (stdout_unit,*) 'Reading ada restart information from ', 'INPUT/'//trim(restart_file_instance)
-       end if
-       call restore_state(ada_driver_restart, DIRECTORY='INPUT')
-    end if
+
+    if (open_file(ada_driver_restart, restart_file, "read", domain, is_restart=.true.)
+      call register_axis(ada_driver_restart, "xaxis_1", size(Atm_var%u_adj,1)
+      call register_axis(ada_driver_restart, "yaxis_1", size(Atm_var%u_adj,2)
+      call register_axis(ada_driver_restart, "zaxis_1", size(Atm_var%u_adj,3)
```

Comment:
Is this a domain decomposed file? If it is then you have to replace `size(Atm_var%u_adj,1)` and `size(Atm_var%u_adj,2)` with "x" and "y", otherwise every rank is going to get the same portion of the data.